lib/AST/Type.cpp

```diff
@@ -1022,6 +1022,31 @@ void ProtocolType::canonicalizeProtocols(
   llvm::array_pod_sort(protocols.begin(), protocols.end(), compareProtocols);
 }
 
+static Type
+getCanonicalInputType(AnyFunctionType *funcType,
+                      std::function<CanType(Type)> getCanonicalType) {
```

Comment:
You can use `llvm::function_ref` instead of `std::function` here, because we're not going to save the function anywhere.

Reply:
Thanks!